### PR TITLE
fix: serverless failing

### PR DESCRIPTION
**Pull request by andreabadesso**
### Motivation

Serverless was failing to package because of multiple issues related to the new yarn workspaces monorepo structure:

1. It wasn't able to find the common module:

```
Error: Cannot find module './@wallet-service/common/package.json'
```

Upon investigation, I found out that this was happening because it yarn workspaces symlinks internal modules to the package's `node_modules` and `serverless-monorepo` was cleaning it

The fix for this issue was adding it as a peer dependency, which actually copied the dependency to the `node_modules` folder and also adding it to be parsed by the `ts-loader`

2. Peer dependencies were not being transpiled by webpack

This is a weird one, but after we solved the issue with the common module, the package was failing in runtime because of errors on the imports

I wasn't to dive deep into why this was the case, but moving the peer dependencies to be dependencies fixed this issue.

My suggestion is to move on with this PR because it is stopping us from fixing a bug affecting our production server (NFT metadata is not being created on S3) and investigate in a KTLO in the near future.

The side effect is only that `yarn install` will take a little longer because it will not hoist the peer dependencies into the root package.

### Acceptance Criteria

- The wallet-service should be able to be packaged and deployed using the common module
- Dependencies should be properly transpiled by webpack

### Checklist
- [X] If you are requesting a merge into `master`, confirm this code is production-ready and can be included in future releases as soon as it gets merged
- [X] Make sure either the unit tests and/or the QA tests are capable of testing the new features
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: .yarnrc.yml [hunk 1]
```diff
@@ -2,6 +2,6 @@ compressionLevel: mixed
 
 enableGlobalCache: false
 
-nmHoistingLimits: dependencies
+nmHoistingLimits: workspaces
```
Comment (pedroferreira1) [inline .yarnrc.yml:6]: Why do you need to change this?

File: packages/wallet-service/src/api/wallet.ts [hunk 2]
```diff
@@ -351,8 +351,7 @@ export const load: APIGatewayProxyHandler = middy(async (event) => {
      */
     await invokeLoadWalletAsync(xpubkeyStr, maxGap);
   } catch (e) {
-    logger.error('Error on lambda wallet invoke', e);
-
+    logger.error(e);
```
Comment (pedroferreira1) [inline packages/wallet-service/src/api/wallet.ts:354]: Why did you change this?

File: packages/wallet-service/webpack.config.js [hunk 3]
```diff
@@ -24,7 +24,9 @@ module.exports = {
     filename: '[name].js',
   },
   target: 'node',
-  externals: [nodeExternals()],
+  externals: [nodeExternals({
+    allowlist: [new RegExp("@wallet-service/common*")],
```
Comment (pedroferreira1) [inline packages/wallet-service/webpack.config.js:33]: Why do you need this?

**Review by pedroferreira1 — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -34,6 +34,7 @@
     "@aws-sdk/client-lambda": "3.540.0",
     "@aws-sdk/client-sqs": "3.540.0",
     "@hathor/wallet-lib": "0.39.0",
+    "@wallet-service/common": "1.5.0",
```
Comment (tuliomir) [inline package.json:37]: question(non-blocking): Will this require us to publish the wallet service common on `npm`? Or will `yarn` be able to identify that this package is not to be taken from an external source?

This is not clear to me because this is the root `package.json`, not a sub module.

**Review by tuliomir — COMMENTED** [review]
suggestion: I agree with your suggestion to move on with this PR and investigate the peer dependencies issue through a KTLO.

My recommendation would be to add the issue number to this PR description now, so that it's easier to track these changes in the future.

File: package.json (hunk 4 above)
Comment (andreabadesso) [reply package.json:37]: It won't, yarn will recognize it is a internal package and symlink it from the package

**Review by tuliomir — APPROVED** [review]

File: packages/wallet-service/src/api/wallet.ts (hunk 2 above)
Comment (andreabadesso) [reply packages/wallet-service/src/api/wallet.ts:354]: That's because Winston as it is configured ignores the second parameter

Created an issue to refactor this: https://github.com/HathorNetwork/hathor-wallet-service/issues/165

File: .yarnrc.yml (hunk 1 above)
Comment (andreabadesso) [reply .yarnrc.yml:6]: Without this setting, the common package is being hoisted in the root node_modules, causing the serverless-monorepo plugin to fail

File: packages/wallet-service/webpack.config.js (hunk 3 above)
Comment (andreabadesso) [reply packages/wallet-service/webpack.config.js:28]: Yarn workspaces hoists modules to the root node_modules directory leaving any incompatible (different semver, etc.) modules with the dependent workspace's node_modules directory, which is the case of our internal package

So we set it as a [external](https://webpack.js.org/configuration/externals/), which is loaded during runtime from the node_modules and not included in the webpack bundle

Comment (andreabadesso) [reply packages/wallet-service/webpack.config.js:33]: The bundle gets too big if we allow webpack to bundle all dependencies so we remove them from the bundle (they [get loaded in runtime](https://webpack.js.org/configuration/externals/))

But we can't do this with the common module, otherwise it would not be seen by the serverless-monorepo package

**Review by pedroferreira1 — APPROVED** [review]
